### PR TITLE
feat(hapi-engine): add schematics

### PR DESCRIPTION
**Pull request by CaerusKaru**
* Adds schematics for the Hapi engine
* Credit to @DevVersion for the Bazel
  infrastructure

### CODE REVIEW

File: modules/hapi-engine/schematics/collection.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "../node_modules/@angular-devkit/schematics/collection-schema.json",
+  "schematics": {
+    "ng-add": {
+      "description": "Adds Angular Universal Hapi Engine to the application without affecting any templates",
+      "factory": "./install",
+      "schema": "./install/schema.json",
+      "aliases": ["express-engine-shell"]
```
Comment (vikerman) [inline modules/hapi-engine/schematics/collection.json]: Change to hapi?

File: modules/hapi-engine/schematics/install/schema.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  /**
+   * The name of the root module class.
+   */
+  rootModuleClassName?: string;
+  /**
+   * Skip installing dependency packages.
+   */
+  skipInstall?: boolean;
+  /**
+   * Skip adding Express server file.
```
[the remark this answers is not in the document]
Comment (vikerman) [reply modules/hapi-engine/schematics/install/schema.ts]: References to express in few other places also

File: modules/hapi-engine/schematics/collection.json (hunk 1 above)
Comment (CaerusKaru) [reply modules/hapi-engine/schematics/collection.json]: Fixed

File: modules/hapi-engine/schematics/install/schema.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (CaerusKaru) [reply modules/hapi-engine/schematics/install/schema.ts]: Fixed

**Comment by CaerusKaru** [comment]
Note: do not merge this until next release is cut (current is 6.0.0). Hapi is not tested extensively and should not be included in a release until it's ready.

**Pull request merged**